### PR TITLE
Fix flaky uao_crash_compaction_row test: part two

### PR DESCRIPTION
**Pull request by pengzhout**
In commit b00916e699c1fa3d95132bdb596e077d87c79cb0, we added a new
    fault injector to wait segment 1 to finish the post cleanup and drop
    the dead ao segment files, however, the test is still flaky because
    segment 1 might delay the dropping of awaiting-drop segment files, this
    is caused by commit 5778f31124d which changed the vacuum behavor:

    "if there are concurrent transactions running, even in READ COMMITTED
    mode, and even if they don't access the table at all, VACUUM will no
    longer be able to recycle compacted segfiles. Next VACUUM will clean
    them up, after the old concurrent transactions."

    in test uao_crash_compaction_row, VACUUM has concurrent transactions.
        1&:VACUUM crash_before_cleanup_phase;
        3:SELECT gp_wait_until_triggered_fault('compaction_before....
    it makes the status of awaiting-drop aoseg files in segment 1
    undetermined.

    To resolve this, we only check the awaiting-drop/compaction aoseg files
    and compaction aoseg files in segment 0 when the first VACUUM (with
    concurrent transactions) is performed, then we do VACUUM again (without
    concurrent transactions) to verify the awaiting-drop/compaction aoseg
    files in segment 1.

    the commit b00916e699 is useless after we changed the test, so revert
    it.

### CODE REVIEW

File: src/backend/commands/vacuum.c [hunk 1]
```diff
@@ -2152,6 +2141,17 @@ vacuum_rel(Oid relid, RangeVar *relation, int options, VacuumParams *params,
...
 	UnlockRelationIdForSession(&onerelid, lmode);
 
+#ifdef FAULT_INJECTOR
+	if (ao_vacuum_phase == VACOPT_AO_POST_CLEANUP_PHASE)
+	{
+		FaultInjector_InjectFaultIfSet(
+			"vacuum_post_cleanup_committed",
+			DDLNotSpecified,
+			"",	// databaseName
+			""); // tableName
```
Comment (gfphoenix78) [inline src/backend/commands/vacuum.c:2151]: Please use SIMPLE_FAULT_INJECTOR instead.

Comment (pengzhout) [reply src/backend/commands/vacuum.c]: this piece of code is abandoned in the latest commits, please take another look.

**Comment by pengzhout** [comment]
> That made me wonder why the same problem does not happen for `crash_before_segmentfile_drop` case?
My analysis is that "VACUUM crash_before_segmentfile_drop" always has a concurrent transaction which is "1&:VACUUM crash_before_cleanup_phase;", so you can see it's output always contain an awaiting-drop aoseg file.
```
1          | 1     | 64  | 1        | 1             | 64               | 2        | 3             | 2
```
> 
> What is the difference between `compaction_before_cleanup_phase` and `compaction_before_segmentfile_drop` faults? They both are defined at effectively the same location in `vacuum_rel()` and are triggered using the same condition... If they are indeed identical, let's collapse themG into one fault and one scenario altogether.

They are used to be different, we used to have two VACUUM phases: AOVAC_DROP and AOVAC_CLEANUP. now 5778f31124d seems to merge them to one phase, so I think there are no different now.

File: src/backend/commands/vacuum.c [hunk 2]
```diff
@@ -1825,7 +1825,6 @@ vacuum_rel(Oid relid, RangeVar *relation, int options, VacuumParams *params,
 		return false;
 	}
 
-#ifdef FAULT_INJECTOR
```
[the remark this answers is not in the document]
Comment (pengzhout) [reply src/backend/commands/vacuum.c:1828]: Yes, fixed, thanks.

**Comment by pengzhout** [comment]
Merged, thanks for the review.

**Pull request merged**